docs/features/tuples.md

```diff
@@ -0,0 +1,227 @@
+}
+```
+
+Note that specifying field names is optional, however when names are provided, all fields must be named. Duplicate names are disallowed.
+
+
+``` c#
+var t1 = (sum: 0, 1);		// error! some fields are named some are not.
+var t2 = (sum: 0, sum:1);	// error! duplicate names.
+```
+
+
+Duality with underlying type
+--------------
+
+Tuples map to underlying types of particular names - 
+
+```
+System.ValueTuple<T1, T2>
+System.ValueTuple<T1, T2, T3>
+...
+System.ValueTuple<T1, T2, T3,..., T7, TRest>
+```
+
+In all scenarios tuple types behave exactly like underlying types with only additional optional enhancement of the more expressive field names given by the programmer.
+
+``` c#
+var t = (sum: 0, count: 1);	
+t.sum   = 1;				// sum   is the name for the field #1 
+t.Item1 = 1;				// Item1 is the name of underlying field #1 and is also avaialable
+
+var t1 = (0, 1);			// tuple omits the field names.
+t.Item1 = 1;				// underlying field name is still avaialable 
+
+t.ToString() 				// ToString on the underlying tuple type is called.
+
+System.ValueTuple<int, int> vt = t;  // identity conversion 
+(int foo, int bar) t2 = vt;  		 // identity conversion
+
+```
+
+Because of the dual nature of tuples, it is not allowed to assign field names that overlap with preexisting member names of the underlying type.
+The only exception is the use of predefined "Item1", "Item2",..."ItemN" at corresponding position N, since that would not be ambiguous.
+
+``` c#
+var t =  (ToString: 0, ObjectEquals: 1);		// error: names match underlying member names
+var t1 = (Item1: 0, Item2: 1);					// valid
+var t2 = (misc: 0, Item1: 1);					// error: "Item1" was used in a wrong position
+```
+
+Example of underlying tuple type implementation:
+https://github.com/dotnet/roslyn/blob/features/tuples/docs/features/ValueTuples.cs
+To be replaced by the actual tuple implementation in FX Core.
+
+
+Identity conversion
+--------------
+
+Element names are immaterial to tuple conversions. Tuples with the same types in the same order are identity convertible to each other or to and from corresponding underlying ValueTuple types, regardless of the names.
+
+``` c#
+var t = (sum: 0, count: 1);	
+
+System.ValueTuple<int, int> vt = t;  // identity conversion 
+(int foo, int bar) t2 = vt;  		 // identity conversion
+
+t2.foo = 1;
+```
+That said, if you have an element name at *one* position on one side of a conversion, and the same name at *another* position on the other side. That would be indication that the code almost certainly contains a bug:
+
+``` c#
+(int sum, int count) foo ()
+{
+	return (count: 1, sum 3); // warning!! 
+}	
+```
+Overloading, Overriding, Hiding
+--------------
+For the purpose of Overloading Overriding Hiding, tuples of the same types and lengths as well as their underlying ValueTuple types are considered equivalent. All other differences are immaterial.
+
+```C#
+class Base 
+{
+	virtual void foo(ValueTuple<int, int> arg){...}
```

Comment:
Avoid `foo` in example documents.  It's a policheck violation by default.  It's okay in this specific context because it's technical content but better to just avoid the violation altogether.